runtime/src/main/java/io/micronaut/health/HeartbeatDiscoveryClientCondition.java

```diff
@@ -35,8 +36,11 @@
 public final class HeartbeatDiscoveryClientCondition implements Condition {
     @Override
     public boolean matches(ConditionContext context) {
-        final CompositeDiscoveryClient compositeDiscoveryClient = context.getBean(CompositeDiscoveryClient.class);
-        final boolean hasDiscovery = compositeDiscoveryClient.getDiscoveryClients().length > 0;
+        final boolean hasDiscovery = context.getBeanContext().getBeanDefinitions(DiscoveryClient.class)
+                .stream()
+                .filter(bd -> bd.getBeanType() != (Class<?>) CompositeDiscoveryClient.class)
```

Comment:
@jameskleeh doesn't the condition always evaluate to `true`? (`CompositeDiscoveryClient` is `abstract`)

Reply:
Yup you are correct @rdesgroppes the check should be for `DefaultCompositeDiscoveryClient`

Reply:
Good catch @rdesgroppes

Reply:
np